src/core/core.interaction.js

```diff
@@ -42,35 +42,32 @@ function evaluateAllVisibleItems(chart, handler) {
 }
 
 /**
- * Helper function to check the items at the hovered index on the index scale
+ * Helper function to get items using binary search, when the data is sorted.
  * @param {Chart} chart - the chart
  * @param {string} axis - the axis mode. x|y|xy
  * @param {object} position - the point to be nearest to
  * @param {function} handler - the callback to execute for each visible item
  * @return whether all scales were of a suitable type
  */
-function evaluateItemsAtIndex(chart, axis, position, handler) {
+function binarySearchItems(chart, axis, position, handler) {
 	const metasets = chart._getSortedVisibleDatasetMetas();
-	const indices = [];
 	for (let i = 0, ilen = metasets.length; i < ilen; ++i) {
 		const metaset = metasets[i];
 		const iScale = metaset.controller._cachedMeta.iScale;
-		if (!iScale || axis !== iScale.axis || !iScale.getIndexForPixel) {
+		if (!iScale || axis !== iScale.axis || !metaset._sorted) {
 			return false;
 		}
-		const index = iScale.getIndexForPixel(position[axis]);
-		if (!isNumber(index)) {
-			return false;
-		}
-		indices.push(index);
 	}
+
 	// do this only after checking whether all scales are of a suitable type
 	for (let i = 0, ilen = metasets.length; i < ilen; ++i) {
-		const metaset = metasets[i];
-		const index = indices[i];
-		const element = metaset.data[index];
-		if (!element.skip) {
-			handler(element, metaset.index, index);
+		const {index, data} = metasets[i];
+		const {lo, hi, loIndex, hiIndex} = lookup(data, axis, position[axis]);
```

Comment:
I don't understand why this is returning two items. Shouldn't the data be located just at one index? It seems like `handler` could be called extra times and after the time scale is split we may not need the "lo" and "hi" concept but then still be using it here